docs/latex/Inhalt/03_Inhalt/04_unit_tests.tex

```diff
@@ -59,8 +64,113 @@ \section{Code-Coverage}
+            .3 21\%.
+            .4 33\%.
+            .4 23\%.
+            .4 0\%.
+            .4 NaN.
+            .4 27\%.
+            .3 81\%.
+        }\end{minipage}
+        &
+        \DTsetlength{0pt}{0pt}{0pt}{0pt}{0pt}
+        \begin{minipage}{2.5cm}\centering\dirtree{%
+            .1 57\%.
+            .2 0\%.
+            .2 66\%.
+            .3 NaN.
+            .3 NaN.
+            .3 66\%.
+            .3 NaN.
+            .3 NaN.
+            .3 NaN.
+            .2 80\%.
+            .3 NaN.
+            .4 NaN.
+            .3 80\%.
+            .4 NaN.
+            .4 77\%.
+            .4 100\%.
+            .2 18\%.
+            .3 33\%.
+            .4 33\%.
+            .4 NaN.
+            .4 NaN.
+            .3 0\%.
+            .4 0\%.
+            .4 0\%.
+            .4 NaN.
+            .4 NaN.
+            .4 0\%.
+            .3 25\%.
+        }\end{minipage} \\
+        \hline
+    \end{tabular}
+    \caption{Code Coverage des gesamten Projektes}
+    \label{tab:code-coverage-full}
+\end{table}
+
 \section{Einsatz von Mocks}
 \label{sec:mocks}
 Mocks spielen eine zentrale Rolle bei der Implementierung von Unit Tests, da sie es ermöglichen, eine Klasse isoliert zu testen. Sie ersetzen dabei die Abhängigkeiten einer Klasse als Objekte mit der für den jeweiligen Test minimal notwendigen Funktionalität. Auch für die im Rahmen dieses Projekts in der Applikationsschicht implementierten Unit Tests kommen Mocks zum Einsatz, welche mit Hilfe des Mocking-Frameworks EasyMock erzeugt werden.
 
-Getestet werden in dieser Schicht der \code{ProfileService} zur Verwaltung der Benutzerprofile, der \code{RecipeService} zur Verwaltung der Rezepte und der \code{MatchingSerivce}, welcher Rezepte mit hoher Übereinstimmung in den Zutaten bestimmt. Alle drei Services besitzen Abhängigkeiten auf mindestens ein Repository für den Zugriff auf Entities und Value Objects. Diese werden für die Unit Tests durch Mocks ersetzt, welche die für den jeweiligen Test notwendigen Daten liefern.
+Getestet werden in dieser Schicht der \code{ProfileService} zur Verwaltung der Benutzerprofile, der \code{RecipeService} zur Verwaltung der Rezepte und der \code{MatchingSerivce}, welcher Rezepte mit hoher Übereinstimmung in den Zutaten bestimmt. Alle drei Services besitzen Abhängigkeiten auf mindestens ein Repository für den Zugriff auf Entities und Value Objects. Diese werden für die Unit Tests durch Mocks ersetzt, welche die für den jeweiligen Test notwendigen Daten liefern. Ein Beispielhafter Test mit einem Mock \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/2-quickie-application/src/test/java/org/pinkcrazyunicorn/quickie/application/profile/ProfileServiceTest.java#L18}{\code{shouldAddProfile}}, in welchem das \code{ProfileRepository} gemockt wird. Dafür wird der Mock zuerst durch EasyMock erstellt, dann trainiert und aktiviert. Abschließend wird geprüft, dass der Mock richtig verwendet wurde.
```

Comment:
"Ein Beispielhafter Test mit einem Mock \code{shouldAddProfile}, in welchem das \code{ProfileRepository} gemockt wird. " -> Müssen wir ganzer Satz? "Ein Beispiel für einen Test, welcher ein Mock verwendet ist \code{shouldAddProfile}"

Reply:
"Dafür wird der Mock zuerst durch EasyMock erstellt, dann trainiert und aktiviert. Abschließend wird geprüft, dass der Mock richtig verwendet wurde." -> "Dazu wird das Mock-Objekt zunächst durch EasyMock erstellt, anschließend trainiert und aktiviert. Abschließend wird überprüft, ob das Mock-Objekt richtig verwendet wurde."